### PR TITLE
Standardize our Powershell code

### PR DESCRIPTION
**Pull request by jaredpar**
In #72341 I made an attempt to standardize our Powershell on pwsh. Unfortunately attempting to get that PR green revealed a number of issues and in following up on those issues with core-eng I came to the conclusion that it's not possible to move fully to pwsh at this time. The issue is that there are downstream consumers of our repo that use Powershell and they have no plans to move hence moving here will just end up breaking source build, unified build, etc ... 

As a result I went through our code and removed all of the existing pwsh usage. I also took the time to update our Powershell Guidelines document to enforce using Powershell as well as a few other items. Then I went through our Powershell scripts and ensured they conformed to these guidelines. 

The one place that remains able to usee pwsh is our VSCode supporting scripts as that is not part of our build infra.

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure PTAL

File: docs/contributing/Powershell Guidelines.md [hunk 1]
```diff
@@ -1,123 +1,95 @@
...
-    [switch]$testDeterminism = $false)
-```
-
-This helps alert callers to casual typos, which would otherwise go unnoticed, by making it an 
-error. 
+The rationale for these parts are:
 
-If the script is complicated enough that it contains a usage / help display, then the following 
-pattern can be used:
+- `Set-StrictMode`: forces forces Powershell into a more strict mode of interpretation and
```
Comment (jjonescz) [inline docs/contributing/Powershell Guidelines.md:50]:
```suggestion
- `Set-StrictMode`: forces Powershell into a stricter mode of interpretation and
```

File: docs/contributing/Powershell Guidelines.md [hunk 2]
```diff
@@ -1,123 +1,95 @@
...
+```powershell
 # DO NOT
-Invoke-Expression "& $command $args"
+& eng/make-bootstrap.ps1
+Write-Host "Done with Bootstrap"
+
 # DO
-Exec-Command $command $args
+& eng/make-bootstrap.ps1
+Test-LastExitCode
```
Comment (jjonescz) [inline docs/contributing/Powershell Guidelines.md:86]: Why not `Exec-Script`?

**Review by jjonescz — APPROVED** [review]

File: docs/contributing/Powershell Guidelines.md (hunk 2 above)
Comment (jaredpar) [reply docs/contributing/Powershell Guidelines.md:86]: It is very difficult to pass switch based values between scripts with `Exec-Script` and Powershell. For example the following is tricky: 

```powershell
Exec-Script "eng/build.ps1" "-ci:$ci" 
```

That turns into `-ci:true` which isn't a valid Powershell calling syntax.

**Comment by jaredpar** [comment]
Not sure why AzDO got disconnected from the PR but here is the build with all passing tests

https://dnceng-public.visualstudio.com/public/_build/results?buildId=593595&view=results

**Pull request merged**